packages/rulesets/src/spectral/functions/arm-path-validation.ts

```diff
@@ -0,0 +1,167 @@
+  ]
+  return matchAnyPatterns(patterns, path)
+}
+
+function verifyResourceType(path: string) {
+  // invalid paths:
+  //  1 <scope>/providers/Microsoft.Compute/{vmName}
+  //  2 <scope>/providers/{resourceName}...
+  //  3 <scope>/providers/ResourceType/...
+  //  4 /providers/Microsoft.Compute/{vmName}
+  const patterns = [/^.*\/providers\/microsoft\.\w+\/\w+.*/gi]
+  return matchAnyPatterns(patterns, path)
+}
+
+function verifyNestResourceType(path: string) {
+  // invalid patterns:
+  // 1 <scope>/providers/Microsoft.Compute/virtualMachine/{vmName}/nestedResourceType/actions
+  // 2 <scope>/providers/Microsoft.Compute/virtualMachine/{vmName}/{nestedResourceType}
+
+  const patterns = [
+    /^.*\/providers\/microsoft\.\w+\/\w+\/{\w+}(?:\/\w+\/(?!default)\w+){1,2}/gi,
+    /^.*\/providers\/microsoft\.\w+(?:\/\w+\/(default|{\w+})){1,2}(?:\/\w+\/(?!default)\w+)+/gi,
+    /^.*\/providers\/microsoft\.\w+\/\w+\/{\w+}(?:\/{\w+})+.*/gi,
+  ]
+  return notMatchPatterns(patterns, path)
+}
+
+function verifyScope(path: string) {
+  // valid patterns:
+  // 1 /<scope>/providers/Microsoft.Compute/virtualMachine/{vmName}
+  // 2 /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine/{vmName}
+  // 3 /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine/{vmName}/providers/Microsoft.Billing/extensions/{extensionName}
+  // 4 /subscriptions/{subscriptionId}/providers/Microsoft.Compute/virtualMachine/{vmName}/providers/Microsoft.Billing/extensions/{extensionName}/{extensionName}
+  const patterns = [
+    /^\/subscriptions\/{subscriptionId}(\/resourceGroups\/{resourceGroupName})?\/providers\/.+/gi,
+    /^\/?{\w+}\/providers\/.+/gi,
+  ]
+  return matchAnyPatterns(patterns, path)
+}
+
+/**
+ *
+ * @param pathKey path key for ARM resource in swagger
+ * @param _opts object containing options for the rule: {
+ *    segmentToCheck: "scope", "resourceGroupParam", "subscriptionIdParam", "resourceType", "nestedResourceType"
+ * }
+ * @param paths
+ * @returns
+ */
+
+export const verifyArmPath = (pathKey: any, _opts: any, paths: any) => {
+  if (pathKey === null || typeof pathKey !== "string") {
+    return []
+  }
+
+  const validOptions = ["scope", "resourceGroupParam", "subscriptionIdParam", "resourceGroupScope", "resourceType", "nestedResourceType"]
+  if (
+    !_opts ||
+    !_opts.segmentToCheck ||
+    (typeof _opts.segmentToCheck === "string" && !validOptions.includes(_opts.segmentToCheck)) ||
+    (Array.isArray(_opts.segmentToCheck) && !_opts.segmentToCheck.every((s: string) => validOptions.includes(s)))
+  ) {
+    return []
+  }
+  const path = paths.path || []
+
+  const errors: any[] = []
+
+  const optionsHandlers = {
+    resourceType: (pathKey: string) => {
+      if (!verifyResourceType(pathKey)) {
+        errors.push({
+          message: `The URI for the CURD methods do not contain a resource type.`,
+          path: [...path],
```

Comment:
I think you can just use
```suggestion
          path,
```